lib/winrm/shells/power_shell.rb

```diff
@@ -34,12 +34,16 @@ def finalize(connection_opts, transport, shell_id)
         end
 
         def close_shell(connection_opts, transport, shell_id)
+          return false unless Thread.current.alive?
+          Thread.current.wakeup if Thread.current.status =~ /sleep/
           msg = WinRM::WSMV::CloseShell.new(
             connection_opts,
             shell_id: shell_id,
             shell_uri: WinRM::WSMV::Header::RESOURCE_URI_POWERSHELL
           )
           transport.send_request(msg.build)
+        rescue ThreadError => e
+          logger.debug("#{e.message}")
```

Comment:
My concern is that this just masks an error that should probably be raised. The underlying transport is still unable to close the shell which could result in orphaned powershell processes on the remote.